datalad/metadata/metadata.py

```diff
@@ -666,8 +666,17 @@ def _get_metadata(ds, types, global_meta=None, content_meta=None, paths=None):
             # we also want to have each unique value set always come
             # in a top-level list, so we known if some unique value
             # was a list, os opposed to a list of unique values
+
+            def _ensure_serializable(val):
+                if isinstance(val, ReadOnlyDict):
+                    return {k: _ensure_serializable(v) for k, v in iteritems(val)}
```

Comment:
Should you use may be val.__class__ here to retain original class which could be derived?

Reply:
I don't think that is necessary.

Reply:
It wouldn't hurt and would make code more scalable happen we come up with derivative classes for ReadOnlyDict (e.g. ReadOnlyOrderedDict) and it would be hard(er) to find all the cases like this.  Let me patch for that and see if that breaks anything (anyways need to merge with master ;-) )

Reply:
or "screw it" - indeed probably not necessary ;)